pkg/config/autodiscover/autodiscover_targets.go

```diff
@@ -66,6 +70,55 @@ func FindTestTarget(labels []configsections.Label, target *configsections.TestTa
 	}
 
 	target.DeploymentsUnderTest = append(target.DeploymentsUnderTest, FindTestDeployments(labels, target, namespace)...)
+	target.Nodes = append(target.Nodes, GetNodesList()...)
+}
+
+// GetNodesList Function that return a list of node and what is the type of them.
+func GetNodesList() (nodes []configsections.Nodes) {
```

Comment:
The intent is to discover all nodes that have workload running from the CNF under test, not the full list of nodes.

Reply:
We already capture node name in the pod data structure. I think we just need to process that and come up with a node list/map. No need to create new handler or invoke new commands. In the future, if we need more info about the nodes, we can invoke get node command then

Reply:
as part of my next task to open debug session will all the nodes, I would need to create "NodeConfig" structure to hold "OC" session. I think I'll get this list of nodes, then add something to indicate if there's any deployment on this node and "OC" session pointer. so it's fine to keep the list of all nodes in here.

Why we want to keep all the nodes?
we do want to test hugepage config and kernel args for all the nodes in the cluster, if CNF scales, we don't know where it will be deployed, hence we need to make sure all nodes are in good state.
Why knowing a deployment is present in node is useful ?
we don't want to drain a node that does not have any deployment.

I would advice to merge this as-is, then work out the details later in my PR.

Reply:
Wouldn't it become a problem when you test a tiny cnf on a huge cluster? I think i am ok with doing platform tests only on the nodes running the workloads. We are not doing run time test. We don't need to worry about whether the whole cluster is setup in the same way. We are testing that the workload does not require non standard node config to run.

Reply:
If we have tiny CNF, we will spend more time testing Nodes that will not be used, but I think that's fine since the tests are not a run time program but rather something that should be run couple of times at most.

Reply:
In all cases, I think it's good to fetch the list of nodes, then filter it based on our need.

Reply:
Ok, we can build the full list as long as there is a way to pick the right one from the list by the tests

Reply:
@hamadise Are you going to update the tests that use node list? If so, Amal's PR can be merged without that part of update.

Reply:
Yes. I'm updating it.